pubsub/subscriptions/subscription_test.go

```diff
@@ -263,6 +264,166 @@ func TestPullMsgsConcurrencyControl(t *testing.T) {
 	}
 }
 
+func TestCreateWithDeadLetterPolicy(t *testing.T) {
+	client := setup(t)
+	ctx := context.Background()
+	tc := testutil.SystemTest(t)
+	deadLetterSourceID := topicID + "-dead-letter-source"
+	deadLetterSubID := subID + "-dead-letter-sub"
+	deadLetterSinkID := topicID + "-dead-letter-sink"
+
+	deadLetterSourceTopic, err := getOrCreateTopic(ctx, client, deadLetterSourceID)
+	if err != nil {
+		t.Fatalf("getOrCreateTopic: %v", err)
+	}
+	defer deadLetterSourceTopic.Delete(ctx)
+	defer deadLetterSourceTopic.Stop()
+
+	deadLetterSinkTopic, err := getOrCreateTopic(ctx, client, deadLetterSinkID)
+	if err != nil {
+		t.Fatalf("getOrCreateTopic: %v", err)
+	}
+	defer deadLetterSinkTopic.Delete(ctx)
+	defer deadLetterSinkTopic.Stop()
+
+	buf := new(bytes.Buffer)
+	if err := createSubWithDeadLetter(buf, tc.ProjectID, deadLetterSubID, deadLetterSourceID, deadLetterSinkTopic.String()); err != nil {
+		t.Fatalf("failed to create a subscription with dead lettering: %v", err)
```

Comment:
Include the name of the thing that failed in the failure message.

https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures

Same with the other failures.